File: types/trial-request.ts
export const DESIRED_CLASSES = [
  "KIDS",
  "ELEMENTARY",
  "MIDDLE_HIGH",
  "ADULT",
] as const;

export type DesiredClass = (typeof DESIRED_CLASSES)[number];

export const DESIRED_CLASS_LABELS: Record<DesiredClass, string> = {
  KIDS: "유치부",
  ELEMENTARY: "초등",
  MIDDLE_HIGH: "중고등",
  ADULT: "성인",
};

export const TRIAL_STATUSES = ["PENDING", "CONFIRMED", "DECLINED"] as const;

export type TrialRequestStatus = (typeof TRIAL_STATUSES)[number];

export const TRIAL_STATUS_LABELS: Record<TrialRequestStatus, string> = {
  PENDING: "대기",
  CONFIRMED: "확정",
  DECLINED: "거절",
};

export type TrialRequest = {
  id: string;
  dojangId: string;
  studentName: string;
  parentName: string;
  parentPhone: string;
  desiredClass: DesiredClass | null;
  memo: string | null;
  status: TrialRequestStatus;
  createdAt: string;
};

File: pumsae_app/lib/features/trials/trial_request.dart
class TrialRequest {
  const TrialRequest({
    required this.id,
    required this.dojangId,
    required this.studentName,
    required this.parentName,
    required this.parentPhone,
    required this.desiredClass,
    required this.memo,
    required this.status,
    required this.createdAt,
  });

  final String id;
  final String dojangId;
  final String studentName;
  final String parentName;
  final String parentPhone;

  /// "KIDS" | "ELEMENTARY" | "MIDDLE_HIGH" | "ADULT" | null, as sent by the
  /// backend.
  final String? desiredClass;
  final String? memo;

  /// "PENDING" | "CONFIRMED" | "DECLINED".
  final String status;
  final DateTime createdAt;

  factory TrialRequest.fromJson(Map<String, dynamic> json) {
    return TrialRequest(
      id: json['id'] as String,
      dojangId: json['dojangId'] as String,
      studentName: json['studentName'] as String,
      parentName: json['parentName'] as String,
      parentPhone: json['parentPhone'] as String,
      desiredClass: json['desiredClass'] as String?,
      memo: json['memo'] as String?,
      status: json['status'] as String,
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }

  TrialRequest copyWith({String? status}) {
    return TrialRequest(
      id: id,
      dojangId: dojangId,
      studentName: studentName,
      parentName: parentName,
      parentPhone: parentPhone,
      desiredClass: desiredClass,
      memo: memo,
      status: status ?? this.status,
      createdAt: createdAt,
    );
  }
}
